code/game/objects/explosion.dm

```diff
@@ -98,23 +98,30 @@ proc/explosion_FX(turf/epicenter, max_range, explosion_sound=get_sfx("explosion"
 	expected_type = /turf
 	flags = EXTENSION_FLAG_IMMEDIATE
 
+//Prevents atoms from being hit by multiple explosion waves at once, and cuts a little bit of cost.
+/atom/movable
+	var/next_explosion_impact = 0
+
 /turf
-	var/next_explosion_ripple = 0
+	var/next_explosion_impact = 0
 
 /turf/proc/explosion_ripple(strength)
 	set waitfor = FALSE
-	if(world.time < next_explosion_ripple)
+	if(world.time < next_explosion_impact)
 		return
-	next_explosion_ripple = world.time + 0.45 SECONDS //Prevents getting spammed by multiple explosions and causing unecessary lag. Yeah this is a little lazy, but I'd rather not assign lists of "processed" turfs or whatever for memory's sake
+	next_explosion_impact = world.time + 0.45 SECONDS //Prevents getting spammed by multiple explosions and causing unecessary lag. Yeah this is a little lazy, but I'd rather not assign lists of "processed" turfs or whatever for memory's sake
 	if(!istype(src, /turf/space))
 		shake_animation(strength)
 	for(var/atom/movable/AM in contents)
+		if(world.time < AM.next_explosion_impact)
+			continue
+		AM.next_explosion_impact = world.time + 0.45 SECONDS
 		AM.ex_act(strength)
 	ex_act(strength)
 
-/datum/extension/explosion/New(atom/parent, devastation_range, heavy_impact_range, light_impact_range, flash_range, adminlog)
+/datum/extension/explosion/New(atom/parent, radius, max_power)
 	.=..()
-	explosion(parent, devastation_range, heavy_impact_range, light_impact_range, flash_range, adminlog)
```

Comment:
shouldnt this be parent.explosion instead?

Extensions are non atomic datums, it can't really create an explosion on its own location

Reply:
See /datum/extension/explosion/proc/explosion

Parent.explosion is the method call that attaches the explosion component, which then subsequently has its own explosion method for handling explosion logic